src/js/modules/accounts/actions.js

```diff
@@ -0,0 +1,142 @@
+import axios from "axios";
+import appHistory from "../../tools/appHistory";
+
+import * as t from "./actionTypes";
+import { STUY_SPEC_API_URL, STUY_SPEC_API_HEADERS } from "../../constants";
+
+export const signUp = registrationParams => {
+  return dispatch => {
+    dispatch({
+      type: t.SIGN_UP_PENDING,
+      payload: registrationParams,
+    });
+
+    const deviseParams = (({ email, password, passwordConfirmation }) => ({
+      email,
+      password,
+      passwordConfirmation,
+    }))(registrationParams);
+    const additionalParams = (({ firstName, lastName }) => ({
+      firstName,
+      lastName,
+    }))(registrationParams);
+
+    axios
+      .post(`${STUY_SPEC_API_URL}/auth`, deviseParams, STUY_SPEC_API_HEADERS)
+      .then(response => {
+        dispatch({
+          type: t.SIGN_UP_FULFILLED,
+          payload: response,
+        });
+        const userId = response.data.data.id;
+        /* As Devise only accepts email/passwords, we need a separate update
+         * for other user properties. Note that we are not simply dispatching
+         * an UPDATE_USER because that will change the form name in the status
+         * of the accounts state slice to editUser, which will prevent success
+         * text from rendering on the signUpForm.
+         */
+        axios.put(
+          `${STUY_SPEC_API_URL}/users/${userId}`,
+          additionalParams,
+          STUY_SPEC_API_HEADERS,
+        );
+      })
+      .catch(err => {
+        dispatch({
+          type: t.SIGN_UP_REJECTED,
+          payload: err,
+        });
+      });
+  };
+};
+
+export const signIn = (signInParams, isInModal) => {
+  return dispatch => {
+    dispatch({
+      type: t.SIGN_IN_PENDING,
+      payload: signInParams,
+    });
+    axios
+      .post(
+        `${STUY_SPEC_API_URL}/auth/sign_in`,
+        signInParams,
+        STUY_SPEC_API_HEADERS,
+      )
+      .then(response => {
+        dispatch({
+          type: t.SIGN_IN_FULFILLED,
+          payload: response,
+        });
+        if (isInModal !== true) {
+          // isInModal may be falsey
```

Comment:
Explain. What are the potential values of isInModal? Shouldn't it be true or false?